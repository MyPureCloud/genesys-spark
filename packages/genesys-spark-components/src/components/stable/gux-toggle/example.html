<div class="container">
  <h1>gux-toggle</h1>
  <gux-toggle checked-label="On" unchecked-label="Off"></gux-toggle>

  <gux-toggle checked-label="On" unchecked-label="Off" checked></gux-toggle>

  <gux-toggle
    checked-label="Active"
    unchecked-label="Inactive"
    label-position="left"
  ></gux-toggle>

  <gux-toggle checked-label="On" unchecked-label="Off" disabled></gux-toggle>

  <gux-toggle
    class="loading"
    checked-label="Active"
    unchecked-label="Inactive"
    loading
  ></gux-toggle>

  <gux-toggle
    class="loading"
    checked-label="Active"
    unchecked-label="Inactive"
    checked
    loading
  ></gux-toggle>

  <gux-toggle
    checked-label="This is another long label for the toggle to show how it works"
    unchecked-label="This is short"
  ></gux-toggle>

  <gux-toggle
    checked-label="On"
    unchecked-label="Off"
    error-message="This is an error message"
  ></gux-toggle>

  <gux-toggle
    checked-label="On"
    unchecked-label="Off"
    label-position="left"
    error-message="This is another error message"
  ></gux-toggle>

  <!-- inline gux-toggle -->
  <div>
    <gux-toggle
      display-inline
      checked-label="On"
      unchecked-label="Off is long"
    ></gux-toggle>
    <gux-toggle
      display-inline
      checked-label="On"
      unchecked-label="Off is long"
      checked
    ></gux-toggle>
  </div>
  <div>
    <gux-toggle
      display-inline
      label-position="left"
      checked-label="On"
      unchecked-label="Off is long"
    ></gux-toggle>
    <gux-toggle
      display-inline
      label-position="left"
      checked-label="On"
      unchecked-label="Off is long"
      checked
    ></gux-toggle>
  </div>
</div>

<style
  onload="(function () {
    Array.from(document.querySelectorAll('.loading'))
      .forEach((element) => {
        setInterval(() => {
          element.loading = Math.random() < 0.8;
        }, 3000)
      });
  })()"
>
  .container {
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    inline-size: 250px;
  }

  gux-toggle {
    margin-block-end: 20px;
  }
</style>
