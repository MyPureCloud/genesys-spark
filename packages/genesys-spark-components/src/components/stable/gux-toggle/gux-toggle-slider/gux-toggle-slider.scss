@use 'sass:math';

$toggle-height: var(--gse-ui-toggle-track-height);
$toggle-width: var(--gse-ui-toggle-track-width);
$slider-diameter: calc(var(--gse-ui-toggle-handle-width));
$slider-transition: calc($toggle-width - $toggle-height);
$slider-transition-duration: 0.2s; // TODO: COMUI-2509 - we are hardcoding this animation duration until design comes back with a token

gux-toggle-slider {
  display: flex;
  align-items: center;
  padding: 2px;
  outline: none;

  .gux-toggle-slider {
    position: relative;
    display: inline-block;
    flex: 0 1 auto;
    align-self: auto;
    order: 0;
    width: $toggle-width;
    height: $toggle-height;
    cursor: pointer;
    border-radius: var(--gse-ui-toggle-track-borderRadius);

    &.gux-error {
      .gux-slider {
        background-color: var(--gse-ui-toggle-track-error-off-backgroundColor);

        .gux-switch {
          border: var(--gse-ui-toggle-handle-error-border-width)
            var(--gse-ui-toggle-handle-error-border-style)
            var(--gse-ui-toggle-handle-error-border-color);
        }
      }

      &.gux-checked {
        .gux-slider {
          background-color: var(--gse-ui-toggle-track-error-on-backgroundColor);
        }
      }
    }

    &.gux-disabled {
      pointer-events: none;
      cursor: default;
      background-color: var(--gse-ui-toggle-track-disabled-off-backgroundColor);
      opacity: var(--gse-ui-toggle-disabled-opacity);
    }

    &:focus-visible {
      border-radius: var(--gse-ui-toggle-focus-borderRadius);
      outline: var(--gse-ui-toggle-focus-border-width)
        var(--gse-ui-toggle-focus-border-style)
        var(--gse-ui-toggle-focus-border-color);
      outline-offset: var(--gse-ui-toggle-focus-offset);
    }

    .gux-slider {
      position: absolute;
      inset: 0;
      background-color: var(--gse-ui-toggle-track-enabled-off-backgroundColor);
      border-radius: calc($toggle-height / 2);
      transition: background-color $slider-transition-duration ease;

      &:hover {
        background-color: var(--gse-ui-toggle-track-hover-off-backgroundColor);
      }

      .gux-switch {
        position: absolute;
        bottom: calc(calc($toggle-height - $slider-diameter) / 2);
        left: calc(($toggle-height - $slider-diameter) / 2);
        width: calc($slider-diameter - 4px);
        height: calc(var(--gse-ui-toggle-handle-height) - 4px);
        color: var(--gse-ui-toggle-handle-foregroundColor);
        content: '';
        background-color: var(--gse-ui-toggle-handle-backgroundColor);
        border: var(--gse-ui-toggle-handle-enabled-border-width)
          var(--gse-ui-toggle-handle-enabled-border-style)
          var(--gse-ui-toggle-handle-enabled-border-color);
        border-radius: var(--gse-ui-toggle-handle-borderRadius);
        transition: $slider-transition-duration;

        gux-icon {
          position: absolute;
          inset: 0;
          width: calc(
            $slider-diameter - 6px
          ); // TODO : COMUI-2528 token needs to be created for 10px as 16px is too big for icon.

          height: calc(
            $slider-diameter - 6px
          ); // TODO : COMUI-2528 token needs to be created for 10px as 16px is too big for icon.

          margin: auto;
          color: var(--gse-ui-toggle-handle-backgroundColor);
          transition: color $slider-transition-duration linear;
        }
      }
    }

    &.gux-checked {
      .gux-slider {
        background-color: var(--gse-ui-toggle-track-enabled-on-backgroundColor);

        &.gux-disabled {
          background-color: var(
            --gse-ui-toggle-track-disabled-on-backgroundColor
          );

          .gux-switch {
            border: var(--gse-ui-toggle-handle-hover-disabled-border-width)
              var(--gse-ui-toggle-handle-hover-disabled-border-style)
              var(--gse-ui-toggle-handle-hover-disabled-border-color);
          }
        }

        &:hover {
          background-color: var(--gse-ui-toggle-track-hover-on-backgroundColor);

          .gux-switch {
            border: var(--gse-ui-toggle-handle-hover-border-width)
              var(--gse-ui-toggle-handle-hover-border-style)
              var(--gse-ui-toggle-handle-hover-border-color);
          }
        }

        .gux-switch {
          transform: translateX($slider-transition);

          gux-icon {
            color: var(--gse-ui-toggle-handle-foregroundColor);
            opacity: 1;
            transition: opacity math.div($slider-transition-duration, 2) linear;
            transition-delay: math.div($slider-transition-duration, 2);
          }
        }
      }
    }
  }
}
