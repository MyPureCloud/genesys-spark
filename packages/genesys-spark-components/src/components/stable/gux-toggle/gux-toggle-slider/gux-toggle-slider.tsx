import { Component, h, JSX, Prop } from '@stencil/core';

@Component({
  styleUrl: 'gux-toggle-slider.scss',
  tag: 'gux-toggle-slider',
  shadow: false
})
export class GuxToggleSlider {
  private checkboxElement: HTMLElement;

  @Prop()
  checked: boolean = false;

  @Prop()
  disabled: boolean = false;

  @Prop()
  guxAriaLabel: string = '';

  @Prop()
  labelId: string = '';

  @Prop()
  errorId: string = '';

  componentDidLoad(): void {
    if (this.labelId) {
      this.checkboxElement?.setAttribute('aria-labelledby', this.labelId);
    } else {
      this.checkboxElement?.setAttribute('aria-label', this.guxAriaLabel);
    }
    if (this.errorId) {
      this.checkboxElement?.setAttribute('aria-describedby', this.errorId);
    }
  }

  render(): JSX.Element {
    const isError = this.errorId.length > 0;

    return (
      <div
        class={{
          'gux-toggle-slider': true,
          'gux-checked': this.checked,
          'gux-disabled': this.disabled,
          'gux-error': isError
        }}
        role="checkbox"
        aria-checked={this.checked.toString()}
        aria-disabled={this.disabled.toString()}
        tabindex={this.disabled ? '' : '0'}
        ref={el => (this.checkboxElement = el)}
      >
        <div class="gux-slider">
          <div class="gux-switch">
            {!isError && (
              <gux-icon icon-name="fa/check-solid" decorative></gux-icon>
            )}
          </div>
        </div>
      </div>
    ) as JSX.Element;
  }
}
