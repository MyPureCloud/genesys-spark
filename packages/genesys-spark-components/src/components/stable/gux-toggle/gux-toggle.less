@import (reference) '../../../style/color.less';
@import (reference) '../../../style/spacing.less';

:host {
  display: block;
  margin: 2px;
  color: @gux-black-50;
  outline: none;
}

:host(.gux-display-inline) {
  display: inline-block;
}

.gux-toggle-container {
  &.gux-disabled {
    pointer-events: none;
    cursor: default;

    .gux-toggle-input {
      .gux-toggle-label {
        .gux-toggle-label-text {
          opacity: 0.5;
        }
      }
    }
  }

  &.gux-toggle-label-left {
    .gux-toggle-input {
      flex-direction: row-reverse;

      .gux-toggle-label {
        .gux-toggle-label-text {
          place-items: end;
        }
      }
    }

    .gux-toggle-error {
      float: right;
    }
  }

  .gux-toggle-input {
    display: flex;
    flex-direction: row;
    flex-wrap: nowrap;
    align-content: stretch;
    align-items: flex-start;
    justify-content: flex-start;
    width: fit-content;
    cursor: pointer;

    .gux-toggle-label {
      position: relative;
      display: inline-block;
      padding: 0 @gux-spacing-2xs;

      .gux-toggle-label-text {
        display: grid;
        grid-template-areas: 'inner-div';
        place-items: start;

        .gux-toggle-label-text-inner {
          position: relative;
          grid-area: inner-div;

          &.gux-hidden {
            visibility: hidden;
          }
        }
      }

      .gux-toggle-label-loading {
        position: absolute;
        inset: 0;
        display: flex;
        flex-direction: row;
        flex-wrap: nowrap;
        align-content: stretch;
        align-items: center;
        justify-content: center;
      }
    }
  }

  .gux-toggle-error {
    .gux-toggle-error-container {
      display: flex;
      flex-direction: row;
      flex-wrap: nowrap;
      align-content: stretch;
      align-items: flex-start;
      justify-content: flex-start;
      font-size: 11px;
      color: @gux-black-50;

      gux-icon {
        flex: 0 1 auto;
        align-self: auto;
        order: 0;
        width: 16px;
        height: 16px;
        margin: 0 4px;
        color: @gux-alert;
      }

      .gux-toggle-error-message {
        flex: 0 1 auto;
        align-self: auto;
        order: 0;
      }
    }
  }
}
