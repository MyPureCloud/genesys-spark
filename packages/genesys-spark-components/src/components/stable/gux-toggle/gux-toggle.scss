:host {
  display: block;
  margin: 2px;
  color: #2e394c;
  outline: none;
}

:host(.gux-display-inline) {
  display: inline-block;
}

.gux-toggle-container {
  &.gux-disabled {
    pointer-events: none;
    cursor: default;

    .gux-toggle-input {
      .gux-toggle-label {
        .gux-toggle-label-text {
          opacity: var(--gse-ui-toggle-disabled-opacity);
        }
      }
    }
  }

  &.gux-toggle-label-left {
    .gux-toggle-input {
      flex-direction: row-reverse;

      .gux-toggle-label {
        .gux-toggle-label-text {
          place-items: end;
        }
      }
    }

    .gux-toggle-error {
      float: right;
    }
  }

  .gux-toggle-input {
    display: flex;
    flex-direction: row;
    flex-wrap: nowrap;
    gap: var(--gse-ui-toggle-gap);
    place-content: stretch flex-start;
    align-items: flex-start;
    width: fit-content;
    cursor: pointer;

    .gux-toggle-label {
      position: relative;
      display: inline-block;

      .gux-toggle-label-text {
        display: grid;
        grid-template-areas: 'inner-div';
        place-items: start;

        .gux-toggle-label-text-inner {
          position: relative;
          grid-area: inner-div;
          font-family: var(--gse-ui-toggle-label-fontFamily);
          font-size: var(--gse-ui-toggle-label-fontSize);
          font-weight: var(--gse-ui-toggle-label-fontWeight);
          color: var(--gse-ui-formControl-label-labelColor);
          word-wrap: break-word;

          &.gux-hidden {
            visibility: hidden;
          }
        }
      }

      .gux-toggle-label-loading {
        position: absolute;
        inset: 0;
        display: flex;
        flex-direction: row;
        flex-wrap: nowrap;
        place-content: stretch center;
        align-items: center;
      }
    }
  }

  .gux-toggle-error {
    .gux-toggle-error-container {
      display: flex;
      flex-direction: row;
      flex-wrap: nowrap;
      gap: var(--gse-ui-toggle-gap);
      place-content: stretch flex-start;
      align-items: flex-start;
      align-items: center;
      padding: var(--gse-ui-formControl-helper-errorPadding);
      font-size: var(--gse-ui-toggle-label-fontSize);
      color: var(--gse-ui-formControl-helper-errorColor);

      gux-icon {
        flex: 0 1 auto;
        align-self: auto;
        order: 0;
        width: var(--gse-ui-button-icon-size);
        height: var(--gse-ui-button-icon-size);
        color: var(--gse-ui-formControl-helper-errorColor);
      }

      .gux-toggle-error-message {
        flex: 0 1 auto;
        align-self: auto;
        order: 0;
      }
    }
  }
}
