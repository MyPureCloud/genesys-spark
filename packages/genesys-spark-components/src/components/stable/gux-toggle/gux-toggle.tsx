import {
  Component,
  Element,
  Event,
  EventEmitter,
  h,
  Host,
  JSX,
  Prop,
  Watch
} from '@stencil/core';

import { trackComponent } from '@utils/tracking/usage';
import { randomHTMLId } from '../../../utils/dom/random-html-id';
import { buildI18nForComponent, GetI18nValue } from '../../../i18n';

import translationResources from './i18n/en.json';

import { GuxToggleLabelPosition } from './gux-toggle.types';

@Component({
  styleUrl: 'gux-toggle.less',
  tag: 'gux-toggle',
  shadow: { delegatesFocus: true }
})
export class GuxToggle {
  private i18n: GetI18nValue;
  private announceElement: HTMLGuxAnnounceBetaElement;
  private labelId: string = randomHTMLId('gux-toggle-label');
  private errorId: string = randomHTMLId('gux-toggle-error');

  @Element()
  private root: HTMLElement;

  @Prop({ mutable: true })
  checked: boolean = false;

  @Prop()
  disabled: boolean = false;

  @Prop()
  loading: boolean = false;

  @Prop()
  checkedLabel: string;

  @Prop()
  uncheckedLabel: string;

  @Prop()
  labelPosition: GuxToggleLabelPosition = 'right';

  @Prop()
  errorMessage: string;

  @Prop()
  displayInline: boolean = false;

  @Watch('loading')
  handleLoading(loading: boolean) {
    if (loading) {
      void this.announceElement.guxAnnounce(this.i18n('toggleIsLoading'));
    } else {
      void this.announceElement.guxAnnounce(
        this.i18n('toggleIsFinishedLoading')
      );
    }
  }

  @Event()
  check: EventEmitter<boolean>;

  private onKeydown(event: KeyboardEvent): void {
    switch (event.key) {
      case ' ':
        event.preventDefault();
        this.toggle();
    }
  }

  private toggle(): void {
    if (!this.disabled && !this.loading) {
      const checkEvent = this.check.emit(!this.checked);
      if (!checkEvent.defaultPrevented) {
        this.checked = !this.checked;
      }
    }
  }

  private getAriaLabel(): string {
    return (
      this.root.getAttribute('aria-label') ||
      this.root.title ||
      this.i18n('defaultAriaLabel')
    );
  }

  async componentWillLoad(): Promise<void> {
    this.i18n = await buildI18nForComponent(this.root, translationResources);

    const variant =
      this.checkedLabel || this.uncheckedLabel ? 'labled' : 'unlabled';

    trackComponent(this.root, { variant });
  }

  private renderLoading(): JSX.Element {
    if (this.loading) {
      return (
        <div class="gux-toggle-label-loading">
          <gux-radial-loading context="input"></gux-radial-loading>
        </div>
      ) as JSX.Element;
    }
  }

  private renderLabel(): JSX.Element {
    if (this.uncheckedLabel && this.checkedLabel) {
      const labelText = this.checked ? this.checkedLabel : this.uncheckedLabel;

      return (
        <div class="gux-toggle-label-and-error">
          <div class="gux-toggle-label">
            <div class="gux-toggle-label-text">
              <span class="gux-toggle-label-text-inner">
                <span id={this.labelId}>{labelText}</span>
                {this.renderLoading()}
              </span>
              <span class="gux-toggle-label-text-inner gux-hidden">
                {this.checkedLabel}
              </span>
              <span class="gux-toggle-label-text-inner gux-hidden">
                {this.uncheckedLabel}
              </span>
            </div>
          </div>
        </div>
      ) as JSX.Element;
    }
  }

  private renderError(): JSX.Element {
    if (this.errorMessage) {
      return (
        <div id={this.errorId} class="gux-toggle-error">
          <div class="gux-toggle-error-container">
            <gux-icon icon-name="alert-warning-octogon" decorative></gux-icon>
            <div class="gux-toggle-error-message">{this.errorMessage}</div>
          </div>
        </div>
      ) as JSX.Element;
    }
  }

  render(): JSX.Element {
    return (
      <Host class={{ 'gux-display-inline': this.displayInline }}>
        <div
          class={{
            'gux-toggle-container': true,
            'gux-toggle-label-left': this.labelPosition === 'left',
            'gux-disabled': this.disabled || this.loading
          }}
        >
          <div class="gux-toggle-input" onClick={this.toggle.bind(this)}>
            <gux-toggle-slider
              checked={this.checked}
              disabled={this.disabled || this.loading}
              guxAriaLabel={this.getAriaLabel()}
              onKeyDown={this.onKeydown.bind(this)}
              labelId={
                this.checkedLabel && this.uncheckedLabel ? this.labelId : ''
              }
              errorId={this.errorMessage ? this.errorId : ''}
            ></gux-toggle-slider>
            {this.renderLabel()}
          </div>
          {this.renderError()}
        </div>
        <gux-announce-beta
          ref={el => (this.announceElement = el)}
        ></gux-announce-beta>
      </Host>
    ) as JSX.Element;
  }
}
