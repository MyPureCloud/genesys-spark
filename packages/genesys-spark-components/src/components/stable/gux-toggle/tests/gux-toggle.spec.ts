import { newSpecPage } from '@test/specTestUtils';
import { GuxToggle } from '../gux-toggle';
import { GuxToggleSlider } from '../gux-toggle-slider/gux-toggle-slider';

const components = [GuxToggle, GuxToggleSlider];
const language = 'en';

describe('gux-toggle', () => {
  it('should build', async () => {
    const page = await newSpecPage({
      components,
      html: `<gux-toggle></gux-toggle>`,
      language
    });

    expect(page.rootInstance).toBeInstanceOf(GuxToggle);
  });

  describe('#render', () => {
    [
      '<gux-toggle></gux-toggle>',
      '<gux-toggle checked></gux-toggle>',
      '<gux-toggle checked disabled></gux-toggle>',
      '<gux-toggle checked-label="On" unchecked-label="Off"></gux-toggle>',
      '<gux-toggle checked checked-label="on" unchecked-label="off"></gux-toggle>',
      `<gux-toggle
        checked-label="On"
        unchecked-label="Off"
        label-position="left"
      ></gux-toggle>`,
      `<gux-toggle
        checked
        checked-label="on"
        unchecked-label="off"
        label-position="right"
      ></gux-toggle>`,
      `<gux-toggle
        checked-label="This is a long label for the toggle to test how it works"
        unchecked-label="This is another long label for the toggle to test how it works"
        label-position="left"
      ></gux-toggle>`,
      `<gux-toggle
        checked
        checked-label="This is a long label for the toggle to test how it works"
        unchecked-label="This is another long label for the toggle to test how it works"
        label-position="right"
      ></gux-toggle>`
    ].forEach((html, index) => {
      it(`should render component as expected (${index + 1})`, async () => {
        const page = await newSpecPage({ components, html, language });

        expect(page.root).toMatchSnapshot();
      });
    });
  });

  describe('User Interactions', () => {
    [
      {
        name: 'clicked',
        userInteraction: (element: HTMLElement) => element.click()
      },
      {
        name: 'Space is pressed',
        userInteraction: (element: HTMLElement) =>
          element.dispatchEvent(new KeyboardEvent('keydown', { key: ' ' }))
      }
    ].forEach(({ name, userInteraction }) => {
      describe(name, () => {
        it(`should not fire a check event when an enabled toggle is disabled and ${name}`, async () => {
          const html =
            '<gux-toggle disabled checked-label="On" unchecked-label="Off"></gux-toggle>';
          const page = await newSpecPage({ components, html, language });
          const element = page.root as HTMLGuxToggleElement;
          const toggleSlider =
            element.shadowRoot.querySelector('gux-toggle-slider');
          const checkSpy = jest.fn();

          element.addEventListener('check', checkSpy);

          userInteraction(toggleSlider);
          await page.waitForChanges();

          expect(checkSpy).not.toHaveBeenCalled();
        });

        it(`should fire a check event when an enabled toggle is ${name}`, async () => {
          const html =
            '<gux-toggle checked-label="On" unchecked-label="Off"></gux-toggle>';
          const page = await newSpecPage({ components, html, language });
          const element = page.root as HTMLGuxToggleElement;
          const toggleSlider =
            element.shadowRoot.querySelector('gux-toggle-slider');
          const checkSpy = jest.fn();

          element.addEventListener('check', checkSpy);

          userInteraction(toggleSlider);
          await page.waitForChanges();

          expect(checkSpy).toHaveBeenCalled();
        });

        it(`should check an unchecked toggle when ${name}`, async () => {
          const html =
            '<gux-toggle checked-label="On" unchecked-label="Off"></gux-toggle>';
          const page = await newSpecPage({ components, html, language });
          const element = page.root as HTMLGuxToggleElement;
          const toggleSlider =
            element.shadowRoot.querySelector('gux-toggle-slider');
          expect(element.checked).toBe(false);

          userInteraction(toggleSlider);
          await page.waitForChanges();

          expect(element.checked).toBe(true);
        });

        it(`should uncheck a checked toggle when ${name}`, async () => {
          const html =
            '<gux-toggle checked checked-label="On" unchecked-label="Off"></gux-toggle>';
          const page = await newSpecPage({ components, html, language });
          const element = page.root as HTMLGuxToggleElement;
          const toggleSlider =
            element.shadowRoot.querySelector('gux-toggle-slider');
          expect(element.checked).toBe(true);

          userInteraction(toggleSlider);
          await page.waitForChanges();

          expect(element.checked).toBe(false);
        });

        it(`should not check an unchecked toggle when disabled and ${name}`, async () => {
          const html =
            '<gux-toggle disabled checked-label="On" unchecked-label="Off"></gux-toggle>';
          const page = await newSpecPage({ components, html, language });
          const element = page.root as HTMLGuxToggleElement;
          const toggleSlider =
            element.shadowRoot.querySelector('gux-toggle-slider');
          expect(element.checked).toBe(false);

          userInteraction(toggleSlider);
          await page.waitForChanges();

          expect(element.checked).toBe(false);
        });

        it(`should not uncheck a checked toggle when disabled and ${name}`, async () => {
          const html =
            '<gux-toggle checked disabled checked-label="On" unchecked-label="Off"></gux-toggle>';
          const page = await newSpecPage({ components, html, language });
          const element = page.root as HTMLGuxToggleElement;
          const toggleSlider =
            element.shadowRoot.querySelector('gux-toggle-slider');
          expect(element.checked).toBe(true);

          userInteraction(toggleSlider);
          await page.waitForChanges();

          expect(element.checked).toBe(true);
        });
      });
    });
  });
});
