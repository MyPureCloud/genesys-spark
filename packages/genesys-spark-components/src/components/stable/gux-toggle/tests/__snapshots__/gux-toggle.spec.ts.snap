// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-toggle #render should render component as expected (1) 1`] = `
<gux-toggle>
  <mock:shadow-root>
    <div class="gux-toggle-container">
      <div class="gux-toggle-input">
        <gux-toggle-slider>
          <div aria-checked="false" aria-disabled="false" aria-label="Toggle Switch" class="gux-toggle-slider" role="checkbox" tabindex="0">
            <div class="gux-slider">
              <div class="gux-switch">
                <gux-icon decorative="" icon-name="fa/check-solid"></gux-icon>
              </div>
            </div>
          </div>
        </gux-toggle-slider>
      </div>
    </div>
    <gux-announce-beta></gux-announce-beta>
  </mock:shadow-root>
</gux-toggle>
`;

exports[`gux-toggle #render should render component as expected (2) 1`] = `
<gux-toggle checked="">
  <mock:shadow-root>
    <div class="gux-toggle-container">
      <div class="gux-toggle-input">
        <gux-toggle-slider>
          <div aria-checked="true" aria-disabled="false" aria-label="Toggle Switch" class="gux-checked gux-toggle-slider" role="checkbox" tabindex="0">
            <div class="gux-slider">
              <div class="gux-switch">
                <gux-icon decorative="" icon-name="fa/check-solid"></gux-icon>
              </div>
            </div>
          </div>
        </gux-toggle-slider>
      </div>
    </div>
    <gux-announce-beta></gux-announce-beta>
  </mock:shadow-root>
</gux-toggle>
`;

exports[`gux-toggle #render should render component as expected (3) 1`] = `
<gux-toggle checked="" disabled="">
  <mock:shadow-root>
    <div class="gux-disabled gux-toggle-container">
      <div class="gux-toggle-input">
        <gux-toggle-slider>
          <div aria-checked="true" aria-disabled="true" aria-label="Toggle Switch" class="gux-checked gux-disabled gux-toggle-slider" role="checkbox" tabindex="">
            <div class="gux-slider">
              <div class="gux-switch">
                <gux-icon decorative="" icon-name="fa/check-solid"></gux-icon>
              </div>
            </div>
          </div>
        </gux-toggle-slider>
      </div>
    </div>
    <gux-announce-beta></gux-announce-beta>
  </mock:shadow-root>
</gux-toggle>
`;

exports[`gux-toggle #render should render component as expected (4) 1`] = `
<gux-toggle checked-label="On" unchecked-label="Off">
  <mock:shadow-root>
    <div class="gux-toggle-container">
      <div class="gux-toggle-input">
        <gux-toggle-slider>
          <div aria-checked="false" aria-disabled="false" aria-labelledby="gux-toggle-label-i" class="gux-toggle-slider" role="checkbox" tabindex="0">
            <div class="gux-slider">
              <div class="gux-switch">
                <gux-icon decorative="" icon-name="fa/check-solid"></gux-icon>
              </div>
            </div>
          </div>
        </gux-toggle-slider>
        <div class="gux-toggle-label-and-error">
          <div class="gux-toggle-label">
            <div class="gux-toggle-label-text">
              <span class="gux-toggle-label-text-inner">
                <span id="gux-toggle-label-i">
                  Off
                </span>
              </span>
              <span class="gux-hidden gux-toggle-label-text-inner">
                On
              </span>
              <span class="gux-hidden gux-toggle-label-text-inner">
                Off
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <gux-announce-beta></gux-announce-beta>
  </mock:shadow-root>
</gux-toggle>
`;

exports[`gux-toggle #render should render component as expected (5) 1`] = `
<gux-toggle checked="" checked-label="on" unchecked-label="off">
  <mock:shadow-root>
    <div class="gux-toggle-container">
      <div class="gux-toggle-input">
        <gux-toggle-slider>
          <div aria-checked="true" aria-disabled="false" aria-labelledby="gux-toggle-label-i" class="gux-checked gux-toggle-slider" role="checkbox" tabindex="0">
            <div class="gux-slider">
              <div class="gux-switch">
                <gux-icon decorative="" icon-name="fa/check-solid"></gux-icon>
              </div>
            </div>
          </div>
        </gux-toggle-slider>
        <div class="gux-toggle-label-and-error">
          <div class="gux-toggle-label">
            <div class="gux-toggle-label-text">
              <span class="gux-toggle-label-text-inner">
                <span id="gux-toggle-label-i">
                  on
                </span>
              </span>
              <span class="gux-hidden gux-toggle-label-text-inner">
                on
              </span>
              <span class="gux-hidden gux-toggle-label-text-inner">
                off
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <gux-announce-beta></gux-announce-beta>
  </mock:shadow-root>
</gux-toggle>
`;

exports[`gux-toggle #render should render component as expected (6) 1`] = `
<gux-toggle checked-label="On" label-position="left" unchecked-label="Off">
  <mock:shadow-root>
    <div class="gux-toggle-container gux-toggle-label-left">
      <div class="gux-toggle-input">
        <gux-toggle-slider>
          <div aria-checked="false" aria-disabled="false" aria-labelledby="gux-toggle-label-i" class="gux-toggle-slider" role="checkbox" tabindex="0">
            <div class="gux-slider">
              <div class="gux-switch">
                <gux-icon decorative="" icon-name="fa/check-solid"></gux-icon>
              </div>
            </div>
          </div>
        </gux-toggle-slider>
        <div class="gux-toggle-label-and-error">
          <div class="gux-toggle-label">
            <div class="gux-toggle-label-text">
              <span class="gux-toggle-label-text-inner">
                <span id="gux-toggle-label-i">
                  Off
                </span>
              </span>
              <span class="gux-hidden gux-toggle-label-text-inner">
                On
              </span>
              <span class="gux-hidden gux-toggle-label-text-inner">
                Off
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <gux-announce-beta></gux-announce-beta>
  </mock:shadow-root>
</gux-toggle>
`;

exports[`gux-toggle #render should render component as expected (7) 1`] = `
<gux-toggle checked="" checked-label="on" label-position="right" unchecked-label="off">
  <mock:shadow-root>
    <div class="gux-toggle-container">
      <div class="gux-toggle-input">
        <gux-toggle-slider>
          <div aria-checked="true" aria-disabled="false" aria-labelledby="gux-toggle-label-i" class="gux-checked gux-toggle-slider" role="checkbox" tabindex="0">
            <div class="gux-slider">
              <div class="gux-switch">
                <gux-icon decorative="" icon-name="fa/check-solid"></gux-icon>
              </div>
            </div>
          </div>
        </gux-toggle-slider>
        <div class="gux-toggle-label-and-error">
          <div class="gux-toggle-label">
            <div class="gux-toggle-label-text">
              <span class="gux-toggle-label-text-inner">
                <span id="gux-toggle-label-i">
                  on
                </span>
              </span>
              <span class="gux-hidden gux-toggle-label-text-inner">
                on
              </span>
              <span class="gux-hidden gux-toggle-label-text-inner">
                off
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <gux-announce-beta></gux-announce-beta>
  </mock:shadow-root>
</gux-toggle>
`;

exports[`gux-toggle #render should render component as expected (8) 1`] = `
<gux-toggle checked-label="This is a long label for the toggle to test how it works" label-position="left" unchecked-label="This is another long label for the toggle to test how it works">
  <mock:shadow-root>
    <div class="gux-toggle-container gux-toggle-label-left">
      <div class="gux-toggle-input">
        <gux-toggle-slider>
          <div aria-checked="false" aria-disabled="false" aria-labelledby="gux-toggle-label-i" class="gux-toggle-slider" role="checkbox" tabindex="0">
            <div class="gux-slider">
              <div class="gux-switch">
                <gux-icon decorative="" icon-name="fa/check-solid"></gux-icon>
              </div>
            </div>
          </div>
        </gux-toggle-slider>
        <div class="gux-toggle-label-and-error">
          <div class="gux-toggle-label">
            <div class="gux-toggle-label-text">
              <span class="gux-toggle-label-text-inner">
                <span id="gux-toggle-label-i">
                  This is another long label for the toggle to test how it works
                </span>
              </span>
              <span class="gux-hidden gux-toggle-label-text-inner">
                This is a long label for the toggle to test how it works
              </span>
              <span class="gux-hidden gux-toggle-label-text-inner">
                This is another long label for the toggle to test how it works
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <gux-announce-beta></gux-announce-beta>
  </mock:shadow-root>
</gux-toggle>
`;

exports[`gux-toggle #render should render component as expected (9) 1`] = `
<gux-toggle checked="" checked-label="This is a long label for the toggle to test how it works" label-position="right" unchecked-label="This is another long label for the toggle to test how it works">
  <mock:shadow-root>
    <div class="gux-toggle-container">
      <div class="gux-toggle-input">
        <gux-toggle-slider>
          <div aria-checked="true" aria-disabled="false" aria-labelledby="gux-toggle-label-i" class="gux-checked gux-toggle-slider" role="checkbox" tabindex="0">
            <div class="gux-slider">
              <div class="gux-switch">
                <gux-icon decorative="" icon-name="fa/check-solid"></gux-icon>
              </div>
            </div>
          </div>
        </gux-toggle-slider>
        <div class="gux-toggle-label-and-error">
          <div class="gux-toggle-label">
            <div class="gux-toggle-label-text">
              <span class="gux-toggle-label-text-inner">
                <span id="gux-toggle-label-i">
                  This is a long label for the toggle to test how it works
                </span>
              </span>
              <span class="gux-hidden gux-toggle-label-text-inner">
                This is a long label for the toggle to test how it works
              </span>
              <span class="gux-hidden gux-toggle-label-text-inner">
                This is another long label for the toggle to test how it works
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <gux-announce-beta></gux-announce-beta>
  </mock:shadow-root>
</gux-toggle>
`;
